tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,779 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const Bluebird = require('bluebird');
+const request = Bluebird.promisifyAll(require('request'));
+const nock = require('nock');
+const s3Packages = require('../../lib/shared/s3-packages');
+
+describe('Shared: s3Packages', function() {
+
+  describe('.getBucketUrlFromVersion()', function() {
+
+    describe('given the version has a short git commit hash build number', function() {
+
+      it('should return the snapshot s3 bucket', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0+6374412');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_SNAPSHOT);
+      });
+
+      it('should return the snapshot s3 bucket if the version has a pre release tag', function() {
+        const bucketUrl = s3Packages.getBucketUrlFromVersion('1.0.0-beta.19+6374412');
+        m.chai.expect(bucketUrl).to.equal(s3Packages.BUCKET_URL_SNAPSHOT);
+      });
+
+    });
+
+    describe('given the version has a short git commit hash build number', function() {
```

Comment:
I don't think so, but I though it would make sense to report that any build number consisting of git commit hashes (short and long), should be detected as snapshot builds. What do you think?